frontend/src/components/MapPanel.tsx

```diff
@@ -110,10 +137,50 @@ function MapPanel({
         map.getCanvas().style.cursor = '';
       });
 
+      const updatedAnnotations = { ...annotations };
+      switch (annotationMode) {
```

Comment:
Small tweak hopefully: Can you break this into its own function called, `changeDrawMode()` ? Since it does not seem to need anything not available to the whole component globally.

Reply:
good timing LOL i was about to be done for merging

Reply:
i think we should change it later because doing it now might make it harder for Sienna to merge back in her changes

Reply:
kk